Develop

Finniki opened the pull request:
# Description

This PR includes all the source codes and images for the UI of the app

Fixes # (issue)

# Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Details

- ### **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

- ### **What is the current behavior?** (You can also link to an open issue here)
User can view and interact with the interface

## Screenshots (if applicable)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

Finniki committed:
add starter files for react project

Finniki committed:
update source files with components

Finniki committed:
update components

Finniki committed:
modify source files and add image folder

gertie-sheshe commented on src/components/SearchBar.css:
```diff
@@ -0,0 +1,42 @@
+  border-radius: 8px;
+  color: #dcdde1;
+  background-color: #184d93;
+  background-color: #0277f3;
```
```suggestion
  background-color: #0277f3;
  border: none:
```

To remove the current default border on the button. Blends better that way

Finniki replied:
Skipped that.. done

gertie-sheshe commented on src/components/searchResults/UserInfo.js:
```diff
@@ -0,0 +1,23 @@
+      />
+      <div className="user-info__intro">
+        <div className="user-info__name">
+          <h1>{props.userInfo.name}</h1>
```
Accessibility catch here :) There'e already a h1 at the top of the page, the next heading level would be h2

Finniki replied:
Of course! skipped again, silly me 😂. I have corrected that

gertie-sheshe commented on src/components/UserDisplay.js:
```diff
@@ -0,0 +1,16 @@
+    </div>
+  );
+};
+
```
```suggestion
const UserDisplay = ({userData, icons}) => {
const { profile, account, contact } = userData;
  return (
    <div className="user-display">
      <UserInfo userInfo={profile} />
      <UserAccount accountInfo={account} />
      <UserContact icons={icons} contactInfo={contact} />
    </div>
  );
};
```


First, how you have it written is correct - and it works ✨. 

We can however improve it by destructuring. This improves the code readability and also you do not have to repeat writing `this.props` often. If this seems complex at first sight - that's alright. You can read more about [props destructuring in React.](https://www.geeksforgeeks.org/destructuring-of-props-in-reactjs/)

Destructuring is a Javascript feature that helps to unpack the content in an object or array. You can read[ more here](https://javascript.info/destructuring-assignment)

Finniki replied:
Yes, I am good with destructuring and it all makes sense.  Effecting that

Finniki replied:
Changed. I one-upped you and did a nested destructuring instead 😝

gertie-sheshe commented on src/components/searchResults/UserContact.js:
```diff
@@ -0,0 +1,39 @@
+    </div>
+  );
+};
+
```
1. These icons and texts should be links, because they represent Twitter profiles, blogs, Github Urls and locations.

2. A way to reduce the code on the file is to have the data in an array (passed down from a parent component) then loop through them here - it would look something like this:

```suggestion
const UserContact = (props) => {
  const iconData = [
    { name: "Lagos", icon: "locationicon" },
    { name: "@me", icon: "twittericon" },
    { name: "email@me", icon: "emailicon" },
  ];
  return (
    <div className="user-contact">
      {iconData.map((data) => (
        <a href={`www.test.com/${data.name}`}>
          {data.name}
          <FontAwesomeIcon icon={data.icon} className="icon" />
        </a>
      ))}
    </div>
  );
};
```

Finniki replied:
Okay. I understand this also and have effected the change.. 
Is test.com just a dead link like #? Seems so, but I can't be sure.

gertie-sheshe commented on src/components/Header.js:
```diff
@@ -0,0 +1,11 @@
+import "./Header.css";
```
First, great work having each component with its own CSS file. Makes it modularized 🙌🏾

However, we still run into the issue of global classes. E.g if you have different components with the same class names, the styles will clash. A solution for this is using [CSS Modules.](https://create-react-app.dev/docs/adding-a-css-modules-stylesheet/) But before that, you can have a read here about the [problem with global css](https://css-tricks.com/css-modules-part-1-need/)

gertie-sheshe reviewed (commented):
This is good work!! @Finniki ✨✨🔥🔥🔥 Thanks for putting in all the effort towards your first React project 🙌🏾

I have added some comments and reviews. Here are other things I noted:
1. Put each component in its own folder. So components folder should have for example Header folder, then inside Header folder, you have the component file and the css file. How you have it is actually fine - but it can become problematic as the files grow.
2. Have a look at [prop-types library ](https://reactjs.org/docs/typechecking-with-proptypes.html)to validate the props in your components.
3. Create a new folder on the same level as components and call it assets and move the images folder there. You can [read this ](https://www.upbeatcode.com/react/where-to-store-images-in-react-app/)to see different approaches for storing images (first section of the article)
4. On smaller screens, the texts come very close together(see screenshot) affecting readability - a solution for this can be to change the font-size of the text on smaller screens.

<img width="480" alt="Screenshot 2022-10-01 at 14 20 43" src="https://user-images.githubusercontent.com/12892109/193407121-6a08ee21-3b47-4aa9-94df-e4f3aa66a96b.png">

All in all - you have done such amazing work!! I know I have given a lot of feedback - we can go through it in a call should you want :)

Finniki committed:
add component folders

Finniki committed:
add assets folder

Finniki committed:
update import statements

Finniki committed:
Delete src/images directory

Finniki committed:
Update Searchbar.css

remove default border on button

Finniki committed:
Update UserInfo.js

Change h1 level header to h2

Finniki committed:
refactor code with destructuring

Finniki committed:
make contact data into links

Finniki committed:
Merge branch 'develop' of github.com:Finniki/github-user-app into develop

gertie-sheshe reviewed (changes requested):
Almost there @Finniki Great job so far! 🙌🏾🔥

1. Good job moving components into their own folders! One more fix would be to use proper naming i.e for `UserAccount` component, the folder name should also be `UserAccount` and not `userAccount`
2. The .css files are still in global scope 😁

Finniki committed:
modify component names to match files

Finniki committed:
change global css to modular css

Finniki commented:
@gertie-sheshe I have implemented the changes.
I encountered an error while trying to rename userInterface folder, it said I couldn't rename it but didn't say why..
Other than that, everything works. I have also implemented CSS modules. 
I have completed the code for the logic as well. Ready to merge this and create a new PR for that

Finniki committed:
change userInterface to Pascal case

Finniki committed:
rename directories with PascalCase

Finniki committed:
update import statements

Finniki committed:
update SearchBar.js file name

Finniki committed:
update source codes

Finniki committed:
update source codes

Finniki committed:
update source code
